protobuf/v1/common.proto

```diff
@@ -13,3 +14,9 @@ message PoolTypeValue {
   PoolType value = 1;
 }
   
+// Protocol over which the bdev/replica/nexus etc will be shared.
+enum ShareProtocol {
+  NONE = 0; // not exposed
```

Comment:
Do we still need this NONE? We can have an optional/oneof ShareProtocol on the replica when it's not shared.

Reply:
So `None` in option will mean that the replica is not shared? IMHO, having the share field as is, and setting an explicit `NONE` value will be better, else it can cause confusions like if the replica is unshared or was the share protocol not set.

Reply:
Yes `None` would mean it's not share, because it doesn't have a protocol at all. 
Ok if that makes more sense to you :+1: In that case maybe we don't need to add the unshare then since you can share with Protocol None..

Reply:
Updated.

Reply:
Ah got it - yes it needs to live in the replica somehow, though you could instead have an optional/oneof ShareProtocol inside the Replica object so it'd become eg:
```rust
Replica {
share_protocol: Option<ShareProtocol>; 
{
```

Reply:
Consider the case of a field, compression. its value can be `off | lz4` etc. It would be better to have to set the value as `compression: Off` rather than having `compression: <None>` right?  I am thinking in similar lines for share protocol also. May be because we used the `NONE` word in share protocol its causing that confusion.

Also in the language guide for [enums in protobuf](https://developers.google.com/protocol-buffers/docs/proto3#enum), I think they have mentioned it similarly, we put the default value as the first one, with 0 value

Reply:
It probably makes sense as part of the replica yes, but maybe not as part of the `ShareReplica`? What does `share_replica` with protocol `NONE` mean?
But anyway this is just a miner thing so if you think this way is better I'm fine with it :+1:

Reply:
One more question on the same thing, What should be the value  of share in CreateReplicaRequest. Should we support NONE there. Its the same as in ShareReplicaRequest, right?

Reply:
Yes I would think so as sometimes you only want to create a local replica so we don't always need to create a replica that is shared.